sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BatchExecUtils.java

```diff
@@ -0,0 +1,113 @@
+    }
+
+    public static String isValid(final List<ItemBatchOperation<?>> operations, final RequestOptions batchOptions) {
+
+        String errorMessage = null;
+
+        if (operations == null) {
+            errorMessage = "expected non-null operations";
+        }
+
+        if (errorMessage == null && operations.size() == 0) {
+            errorMessage = "expected operations.size > 0";
+        }
+
+        if (errorMessage == null && batchOptions != null) {
+          if (batchOptions.getIfMatchETag() != null || batchOptions.getIfNoneMatchETag() != null) {
+              errorMessage = "one or more request options provided on the batch request are not supported";
+          }
+        }
+
+        if (errorMessage == null) {
+            for (ItemBatchOperation<?> operation : operations) {
+
+                final RequestOptions batchOperationOptions = operation.getRequestOptions();
+                final Map<String, Object> batchOperationProperties = batchOperationOptions != null ? batchOperationOptions.getProperties() : null;
+
+                if (batchOperationProperties != null
+                    && (batchOperationProperties.containsKey(WFConstants.BackendHeaders.EFFECTIVE_PARTITION_KEY)
+                    || batchOperationProperties.containsKey(WFConstants.BackendHeaders.EFFECTIVE_PARTITION_KEY_STRING)
+                    || batchOperationProperties.containsKey(HttpConstants.HttpHeaders.PARTITION_KEY))) {
+
+                    final String epkString = (String) batchOperationProperties.computeIfPresent(
+                        WFConstants.BackendHeaders.EFFECTIVE_PARTITION_KEY_STRING,
+                        (k, v) -> v instanceof String ? v : null);
+
+                    final byte[] epk = (byte[]) batchOperationProperties.computeIfPresent(
+                        WFConstants.BackendHeaders.EFFECTIVE_PARTITION_KEY,
+                        (k, v) -> v instanceof byte[] ? v : null);
+
+                    final String pkString = (String) batchOperationProperties.computeIfPresent(
+                        HttpConstants.HttpHeaders.PARTITION_KEY,
+                        (k, v) -> v instanceof String ? v : null);
+
+                    if ((epk == null && pkString == null) || epkString == null) {
+                        return lenientFormat(
+                            "expected byte[] value for %s and string value for %s, not (%s, %s)",
+                            WFConstants.BackendHeaders.EFFECTIVE_PARTITION_KEY,
+                            WFConstants.BackendHeaders.EFFECTIVE_PARTITION_KEY_STRING,
+                            epk == null
+                                ? (pkString == null ? "null" : pkString)
+                                : ByteBufUtil.hexDump(epk), epkString == null ? "null" : epkString);
+                    }
+
+                    if (operation.getPartitionKey() != null ) {
+                        errorMessage = "partition key and effective partition key may not both be set.";
```

Comment:
Agreed. My fault. Not likely. Since Java has specific request options and they don't extent request options class which has this property that can have all kind of header set, we don't need to validate this much here.  Will make it shorter.

Reply:
Remove these functions. They are of no use with no property map.